File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.go
func isValid(s string) bool {
	stack := []string{}
	m := map[string]string{
		"]": "[",
		"}": "{",
		")": "(",
	}

	for i := 0; i < len(s); i++ {
		if string(s[i]) == "(" || string(s[i]) == "[" || string(s[i]) == "{" {
			stack = append(stack, string(s[i]))
			continue
		}
		if len(stack) == 0 {
			return false
		}
		c := stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		if m[string(s[i])] != string(c) {
			return false
		}
	}
	return len(stack) == 0
}

File: LeetCode/Easy/0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    const obj = {
        ")": "(",
        "]": "[",
        "}": "{"
    }
    const arr = []

    for (let i = 0; i < s.length; i++) {
        if (!obj.hasOwnProperty(s[i])) {
            arr.push(s[i])
            continue
        }
        else if (obj[s[i]] !== arr.pop()) {
            return false
        }
    }
    return arr.length == 0
};